packages/hardhat-core/src/internal/hardhat-network/provider/modules/eth.ts

```diff
@@ -1318,47 +1349,138 @@ export class EthModule {
     this._logger.log(util.inspect(error));
   }
 
-  private _logFrom(from: Buffer) {
-    this._logger.logWithTitle("From", bufferToHex(from));
+  private _logFrom(from: Buffer, indent = false) {
+    this._logger.logWithTitle("From", bufferToHex(from), indent);
   }
 
   private async _sendTransactionAndReturnHash(tx: Transaction) {
```

Comment:
This method is way to long and needs to be refactored. Let's group the operations and extract functions like:
* log block
  * log a transaction from block
* log the currently sent transaction